From 80e309c0f0b06d5ba24072976109b7686192d92a Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Fri, 31 Dec 2021 15:59:17 +0800
Subject: [PATCH] kabi: reserve space for net_namespace
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 80e309c0f0b06d5ba24072976109b7686192d92a


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KKML
DTS: NA
CVE: NA

---------------------------------------------------------

Reserve space in net_namespace.h

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/net/net_namespace.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/net/net_namespace.h b/include/net/net_namespace.h
index eb0e7731f3b1..c7faca9d7447 100644
--- a/include/net/net_namespace.h
+++ b/include/net/net_namespace.h
@@ -38,6 +38,7 @@
 #include <linux/idr.h>
 #include <linux/skbuff.h>
 #include <linux/notifier.h>
+#include <linux/kabi.h>
 
 struct user_namespace;
 struct proc_dir_entry;
@@ -189,6 +190,11 @@ struct net {
 	struct sock		*crypto_nlsk;
 #endif
 	struct sock		*diag_nlsk;
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 } __randomize_layout;
 
 #include <linux/seq_file_net.h>
-- 
2.26.2

